// Copyright 2022 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message ClouddeployDeliveryPipeline {
  string name = 1;
  string uid = 2;
  string description = 3;
  map<string, string> annotations = 4;
  map<string, string> labels = 5;
  string create_time = 6;
  string update_time = 7;
  ClouddeployDeliveryPipelineSerialPipeline serial_pipeline = 8;
  ClouddeployDeliveryPipelineCondition condition = 9;
  string etag = 10;
  string project = 11;
  string location = 12;
  bool suspended = 13;
}

message ClouddeployDeliveryPipelineSerialPipeline {
  repeated ClouddeployDeliveryPipelineSerialPipelineStages stages = 1;
}

message ClouddeployDeliveryPipelineSerialPipelineStages {
  string target_id = 1;
  repeated string profiles = 2;
  ClouddeployDeliveryPipelineSerialPipelineStagesStrategy strategy = 3;
}

message ClouddeployDeliveryPipelineSerialPipelineStagesStrategy {
  ClouddeployDeliveryPipelineSerialPipelineStagesStrategyStandard standard = 1;
}

message ClouddeployDeliveryPipelineSerialPipelineStagesStrategyStandard {
  bool verify = 1;
}

message ClouddeployDeliveryPipelineCondition {
  ClouddeployDeliveryPipelineConditionPipelineReadyCondition pipeline_ready_condition = 1;
  ClouddeployDeliveryPipelineConditionTargetsPresentCondition targets_present_condition = 2;
}

message ClouddeployDeliveryPipelineConditionPipelineReadyCondition {
  bool status = 1;
  string update_time = 2;
}

message ClouddeployDeliveryPipelineConditionTargetsPresentCondition {
  bool status = 1;
  repeated string missing_targets = 2;
  string update_time = 3;
}

message ApplyClouddeployDeliveryPipelineRequest {
  ClouddeployDeliveryPipeline resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteClouddeployDeliveryPipelineRequest {
  string service_account_file = 1;
  ClouddeployDeliveryPipeline resource = 2;
}

message ListClouddeployDeliveryPipelineRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListClouddeployDeliveryPipelineResponse {
  repeated ClouddeployDeliveryPipeline items = 1;
}

service ClouddeployDeliveryPipelineService {
  rpc ApplyClouddeployDeliveryPipeline(ApplyClouddeployDeliveryPipelineRequest) returns (ClouddeployDeliveryPipeline);
  rpc DeleteClouddeployDeliveryPipeline(DeleteClouddeployDeliveryPipelineRequest) returns (google.protobuf.Empty);
  rpc ListClouddeployDeliveryPipeline(ListClouddeployDeliveryPipelineRequest) returns (ListClouddeployDeliveryPipelineResponse);
}
